### PR TITLE
Remove config prompting for secrets and text

### PR DESCRIPTION
**Pull request by hub-cap**
This commit removes the ability to use ${prompt.secret} and
${prompt.text} as valid config settings. Secure settings has obsoleted
the need for this, and it cleans up some of the code in Bootstrap.

### CODE REVIEW

**Comment by hub-cap** [comment]
I chatted w/ @rjernst and i will be finding a way to add deprecation logging to this. I dont know how early this is in bootstrap but i suspect its before logging is set up, but ill find a way to hack it in.

File: core/src/main/java/org/elasticsearch/node/InternalSettingsPreparer.java [hunk 1]
```diff
@@ -128,10 +122,25 @@ static void initializeSettings(final Settings.Builder output, final Settings inp
     }
 
     /**
-     * Finish preparing settings by replacing forced settings, prompts, and any defaults that need to be added.
-     * The provided terminal is used to prompt for settings needing to be replaced.
+     * Checks all settings values to make sure they do not have the old prompt settings. These were deprecated in 6.0.0.
      */
-    private static void finalizeSettings(Settings.Builder output, Terminal terminal) {
+    private static void checkSettingsForTerminalDeprecation(final Settings.Builder output) throws SettingsException {
+        for (String setting : output.keys()) {
```
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/node/InternalSettingsPreparer.java]: Let's add an assertion here the major property of the current version is not 8; then, when we bump the version to 8.0.0 on master we know that we can remove this code.

**Review by jasontedor — APPROVED** [review]
I left one comment, please address before merging. Otherwise, I'm so happy to see this simplification. LGTM.

File: core/src/main/java/org/elasticsearch/node/InternalSettingsPreparer.java (hunk 1 above)
Comment (hub-cap) [reply core/src/main/java/org/elasticsearch/node/InternalSettingsPreparer.java]: dude, nice. Thats smart. Will do.

**Comment by jasontedor** [comment]
> I chatted w/ @rjernst and i will be finding a way to add deprecation logging to this (where this is the 6.x version of this). I dont know how early this is in bootstrap but i suspect its before logging is set up, but ill find a way to hack it in.

There's a way to hack it in: pass whether or not the terminal was used to set a setting up through the environment and then in bootstrap extract that from the environment and log a deprecation message if it was used. Although, I do wonder if this is overkill (is emitting a message on the terminal enough?)?

**Comment by hub-cap** [comment]
> Although, I do wonder if this is overkill 

I think it would be nice to have in the deprecation log. It can cause peoples clusters to break and they wont really know it until spinning them up if they pay less attn to the standard log / terminal. That said, is there any prior art to what we do in these infra-y breaking changes? If we do not normally do this, then its perfectly valid not to add it for this. Im still leaning toward seeing it in the deprecation logs tho.

**Comment by jasontedor** [comment]
> That said, is there any prior art to what we do in these infra-y breaking changes?

There are times where we simply can not have the fancy deprecation logging and rely on the migration docs. I think I want to see how ugly and invasive the hack is before rendering judgement here. My concern with a hack that lives in 6.x only is it making backporting other changes more difficult.

**Comment by hub-cap** [comment]
Ill give it a whirl and see how ugly it is.

File: core/src/main/java/org/elasticsearch/node/InternalSettingsPreparer.java [hunk 2]
```diff
@@ -123,8 +124,13 @@ static void initializeSettings(final Settings.Builder output, final Settings inp
 
     /**
      * Checks all settings values to make sure they do not have the old prompt settings. These were deprecated in 6.0.0.
+     * This check should be removed in 8.0.0.
      */
     private static void checkSettingsForTerminalDeprecation(final Settings.Builder output) throws SettingsException {
+        // This method to be removed in 8.0.0, as it was deprecated in 6.0 and removed in 7.0
+        if (Version.CURRENT.toString().startsWith("8")) {
```
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/node/InternalSettingsPreparer.java]: How about `Version.CURRENT.major == 8`? And I think maybe we should only make this an assertion?

**Review by jasontedor — CHANGES_REQUESTED** [review]
I left a comment about the assertion.

File: core/src/main/java/org/elasticsearch/node/InternalSettingsPreparer.java (hunk 2 above)
Comment (hub-cap) [reply core/src/main/java/org/elasticsearch/node/InternalSettingsPreparer.java]: Thx, i wasnt aware exactly how to do it

**Review by jasontedor — APPROVED** [review]
LGTM.

**Pull request merged**